### PR TITLE
Add Dockerfile to main repository

### PR DESCRIPTION
**Pull request by so0k**
I suggest to include Dockerfile to main repository.

If you agree, I will remove the references to the honestbee image on quay.io and you can add references to the official docker registry you want to push the builds to.

Edit: Checklist
- [x] Rebase on official python (alpine)
- [x] Add tini process reaper
- [ ] Add option to entrypoint bash script to attach the send-alert process to uwsgi daemon
- [ ] Verify all assets are served properly 
  I noticed some admin UI buttons seemed to have gone missing, could be because of asset compilation needs hostname?
- [ ] Find a way to get rid of `chown -R hc:hc /usr/src/app`
- [ ] reconsider using `gosu`

### CODE REVIEW

**Comment by so0k** [comment]
this is alpine based and production ready image.

Refer to https://github.com/haswalt/docker-healthchecks/issues/5

**Comment by so0k** [comment]
As discussed on the other thread I'll add the features requested:

- [x] Rebase on official python (alpine)
- [x] Add tini process reaper
- [ ] Add option to entrypoint bash script to attach the send-alert process to uwsgi daemon
- [ ] Verify all assets are served properly (I noticed some admin UI buttons seemed to have gone missing (could be because of asset compilation needs hostname?
- [ ] Find a way to get rid of `chown -R hc:hc /usr/src/app`
- [ ] reconsider using `gosu`

I'll edit these into the main description of the PR

**Comment by cuu508** [comment]
Hi @so0k, I'm hesitant to add deployment bits to the main repository. A couple reasons:

* everyone does deployment differently. Fabric, Ansible playbooks, Docker, Elastic Beanstalk etc. 
* even in the case of Docker, it would be hard to have a Dockerfile that is 1) concise 2) works for everyone. As an example, I'm looking into using Dokku (work in progress here: https://github.com/cuu508/dokku-healthchecks). I've added a CHECKS file and am adding a Procfile which would be useful in Dokku evironment but would not make sense outside it. 

So in short I would prefer to put infrastructure & deployment bits in separate repositories.

**Comment by haswalt** [comment]
@cuu508 @so0k I'm happy to keep my version maintained again. I'm working on improving the docker image at the moment. Maybe we add links to the readme with deployment instructions?

**Comment by szepeviktor** [comment]
> I'm hesitant to add deployment bits to the main repository

A Wiki is nice place to a single file (or two)

**Comment by arthurlogilab** [comment]
@cuu508 I would argue in favor of including a Dockerfile (and other means of deploy) in the repo. When looking around for a project, even if when I deploy in production with tool X, I find it a sign of quality for find a Dockerfile, or/and a docker-compose.yml, a Vagrantfile etc. This tells me "hey! you can test be real quick even if you end up deploying with your tool of choice.

**Comment by szepeviktor** [comment]
> This tells me "hey! you can test be real quick even if you end up deploying with your tool of choice.

For me it tells: "hey! WE can test our product real quick" so it breaks before users install it.

**Comment by cristaloleg** [comment]
bump

**Comment by varac** [comment]
bump

**Comment by cuu508** [comment]
I've now added (experimental for now) Dockerfile and docker-compose.yml – https://github.com/healthchecks/healthchecks/tree/master/docker
